reactor-core/src/main/java/reactor/core/scheduler/Schedulers.java

```diff
@@ -118,9 +118,24 @@ public static Scheduler fromExecutor(Executor executor, boolean trampoline) {
 	 * @param executorService an {@link ExecutorService}
 	 *
 	 * @return a new {@link Scheduler}
+	 * @deprecated prefer using {@link #fromExecutorService(ExecutorService, String)},
+	 * especially if you plan on using metrics as this gives the executor a meaningful identifier
 	 */
+	@Deprecated
 	public static Scheduler fromExecutorService(ExecutorService executorService) {
-		return new DelegateServiceScheduler(executorService);
+		return fromExecutorService(executorService, null);
```

Comment:
WDYT about using `"anonymous-" + System.identityHashCode(executorService)`?

Reply:
good idea! plus added benefit of avoiding nullable string 👍

Reply:
updated